### PR TITLE
[wasm] Javascript Core Function

### PR DESCRIPTION
**Pull request by kjpou1**
Support JavaScript [Function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function).

- Create a new Function:
``` csharp
var sum = new Function("a", "b", "return a + b");
Console.WriteLine($"Sum: {sum.Call(null, 3, 5)}");
```
```
console output ---

Sum: 8
```

- Call
- Apply
- Bind

- Marshalling of Functions passed from JavaScript to C#

``` csharp
            var math = (JSObject)Runtime.GetGlobalObject("Math");
            var min = (Function)math.GetObjectProperty("min");
            Console.WriteLine($"Min: {min.Apply(null, new object[] { 5, 6, 2, 3, 7 })}");
```
```
console output ---

Min: 2
```

- Add tests to binding-tests.cs.

- Fix up the TypedArray tests also included.

### CODE REVIEW

File: sdks/wasm/driver.c [hunk 1]
```diff
@@ -761,6 +761,18 @@ mono_wasm_get_obj_type (MonoObject *obj)
 	}
 }
 
+EMSCRIPTEN_KEEPALIVE int
+mono_wasm_is_simple_array (MonoObject *obj)
+{
```
Comment (vargaz) [inline sdks/wasm/driver.c]: You can do if (a is Array arr) { if (arr.Rank == 1 && a.GetLowerdBound (0) == 0) ... }

Comment (kjpou1) [reply sdks/wasm/driver.c]: @vargaz I can do `if (a is Array arr)` in a `.c` program?  That looks more like c-sharp.

Comment (vargaz) [reply sdks/wasm/driver.c]: I mean the calling c# code can do that instead of adding an icall.

**Comment by lewing** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build failed

File: sdks/wasm/driver.c (hunk 1 above)
Comment (kjpou1) [reply sdks/wasm/driver.c]: Address icall in recent commit.

**Comment by kjpou1** [comment]
@monojenkins build failed

**Pull request merged**